tests/test_upload_commands.py

```diff
@@ -0,0 +1,200 @@
+        assert command.param3 == 0
+        assert command.param4 == 0
+        assert abs(command.x - lat_input) <= TOLERANCE 
+        assert abs(command.y - lon_input) <= TOLERANCE
+        assert command.z == altitude_input
+
+
+def test_upload_empty_command_list(drone: dronekit.Vehicle,
+                                   commands: "list[dronekit.Command]") -> None:
+    """
+    Test the case of an empty command list.
+    """
+    upload_commands.upload_commands(drone,commands)
+
+    # Retrieve current drone commands and see if they match with previous inputs
+    command_sequence = drone.commands 
+    command_sequence.download()
+    drone.wait_ready()
+
+    for i, command in enumerate(command_sequence):
+        lat_input = waypoints_input[i][0]
+        lon_input = waypoints_input[i][1]
+
+        assert command.frame == MAVLINK_FRAME
+        assert command.command == MAVLINK_COMMAND
+        assert command.param1 == DELAY
+        # Parameters 2,3,4 not being tested since Mission Planner ignores them
+        assert command.param2 == 0
+        assert command.param3 == 0
+        assert command.param4 == 0
+        assert abs(command.x - lat_input) <= TOLERANCE
+        assert abs(command.y - lon_input) <= TOLERANCE
+        assert command.z == altitude_input
+
+        
+def test_upload_takeoff_command(drone: dronekit.Vehicle,
+                                 commands: "list[dronekit.Command]") -> None:
+    """
+    Test the case of a takeoff command.
+    """
+    upload_commands.upload_commands(drone,commands)
+
+    # Retrieve current drone commands and see if they match with previous inputs
+    command_sequence = drone.commands 
+    command_sequence.download()
+    drone.wait_ready()
+
+    for command in command_sequence:
+        assert command.frame == MAVLINK_FRAME
+        assert command.command == MAVLINK_TAKEOFF
+        assert command.param1 == 0
+        assert command.param2 == 0
+        assert command.param3 == 0
+        assert command.param4 == 0
+        assert command.x == 0
+        assert command.y == 0
+        assert command.z == altitude_input
+    
+
+def test_upload_landing_command(drone: dronekit.Vehicle,
+                                 commands: "list[dronekit.Command]") -> None:
+    """
+    Test the case of a landing command.
+    """
+    upload_commands.upload_commands(drone,commands)
+
+    # Retrieve current drone commands and see if they match with previous inputs
+    command_sequence = drone.commands 
+    command_sequence.download()
+    drone.wait_ready()
+
+    for command in command_sequence:
+        assert command.frame == MAVLINK_FRAME
+        assert command.command == MAVLINK_LANDING
+        assert command.param1 == 0
+        assert command.param2 == 0
+        assert command.param3 == 0
+        assert command.param4 == 0
+        assert command.x == 0
+        assert command.y == 0
+        assert command.z == 0
+
+    
+if __name__ == "__main__":

```

Comment:
Since the test function is being generalized, you can call it repeatedly with a different list of commands each time.